tests/neuro_san/client/keywords_agent_evaluator.py

```diff
@@ -10,50 +10,21 @@
 # END COPYRIGHT
 
 from typing import Any
-from typing import List
 
-from neuro_san.message_processing.basic_message_processor import BasicMessageProcessor
+from tests.neuro_san.client.abstract_agent_evaluator import AbstractAgentEvaluator
 
-from tests.neuro_san.client.agent_evaluator import AgentEvaluator
-from tests.neuro_san.client.assert_forwarder import AssertForwarder
 
-
-class KeywordsAgentEvaluator(AgentEvaluator):
+class KeywordsAgentEvaluator(AbstractAgentEvaluator):
     """
-    AgentEvaluator implementation that looks for specific keywords in output.
+    AbstractAgentEvaluator implementation that looks for specific keywords in output.
     """
 
-    def __init__(self, asserts: AssertForwarder, test_key: str):
-        """
-        Constructor
-
-        :param asserts: The AssertForwarder instance to handle failures
-        :param test_key: The chat message key to look for output
-        """
-        self.asserts: AssertForwarder = asserts
-        self.test_key: str = test_key
-
-    def evaluate(self, processor: BasicMessageProcessor, verify_for: Any):
+    def test_one(self, verify_value: Any, test_value: Any):
         """
-        Evaluate the contents of the BasicMessageProcessor
-
-        :param processor: The BasicMessageProcessor to evaluate
-        :param verify_for: The data to evaluate the response against
+        :param verify_value: The value to verify against
+        :param test_value: The value appearing in the test sample
         """
-        test_me: str = None
-
-        if self.test_key == "text":
-            test_me = processor.get_answer()
-        elif self.test_key == "sly_data":
-            # Not yet
-            return
-
-        self.asserts.assertIsNotNone(test_me)
-
-        verify_all: List[str] = verify_for
-        if isinstance(verify_for, str):
-            verify_all = [verify_for]
-        self.asserts.assertIsInstance(verify_all, List)
-
-        for verify_one in verify_all:
-            self.asserts.assertIn(verify_one, test_me)
```

Comment:
All of the above logic gets abstracted away and vamped on in AbstractAgentEvaluator.